kotlinpoet/src/main/java/com/squareup/kotlinpoet/FileSpec.kt

```diff
@@ -248,7 +248,7 @@ public class FileSpec private constructor(
      * The annotation must either have a [`file` use-site target][AnnotationSpec.UseSiteTarget.FILE]
      * or not have a use-site target specified (in which case it will be changed to `file`).
      */
-    public fun addAnnotation(annotationSpec: AnnotationSpec): Builder = apply {
+    override fun addAnnotation(annotationSpec: AnnotationSpec): Builder = apply {
```

Comment:
This override can be bypassed by mutating the collection directly. Should probably move the validation to the builder, but that's a separate issue.